sqlite.go

```diff
@@ -377,8 +422,35 @@ func (r *rows) Next(dest []driver.Value) (err error) {
 				if err != nil {
 					return err
 				}
-
+				cty, err := r.columnDeclType(i)
```

Comment:
This and other related changes have to be executed only in compatibility mode but I can't find the if/switch guard.